node/pkg/reporter/reporter_test.go

```diff
@@ -1,607 +1,130 @@
-// 	if err != nil {
-// 		t.Fatalf("error creating chain helper: %v", err)
-// 	}
-// 	defer tmpHelper.Close()
-
-// 	whitelist, err := ReadOnchainWhitelist(ctx, tmpHelper, contractAddress, GET_ONCHAIN_WHITELIST)
-// 	if err != nil {
-// 		t.Fatalf("error reading onchain whitelist: %v", err)
-// 	}
-// 	reporter, err := NewReporter(
-// 		ctx,
-// 		WithHost(testItems.app.Host),
-// 		WithPubsub(testItems.app.Pubsub),
-// 		WithConfigs(submissionPairs),
-// 		WithInterval(5000),
-// 		WithContractAddress(contractAddress),
-// 		WithCachedWhitelist(whitelist),
-// 		WithJobType(DeviationJob),
-// 	)
-// 	if err != nil {
-// 		t.Fatalf("error creating new deviation reporter: %v", err)
-// 	}
-
-// 	err = reporter.deviationJob()
-// 	if err != nil {
-// 		t.Fatal("error running deviation job")
-// 	}
-// }
+import (
+	"context"
+	"os"
+	"testing"
+	"time"
+
+	"bisonai.com/orakl/node/pkg/chain/helper"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestNewReporter(t *testing.T) {
+	ctx := context.Background()
+	app := New()
+
+	submissionPairs, err := getConfigs(ctx)
+	if err != nil {
+		t.Fatalf("error getting submission pairs: %v", err)
+	}
+	groupedSubmissionPairs := groupConfigsBySubmitIntervals(submissionPairs)
+
+	contractAddress := os.Getenv("SUBMISSION_PROXY_CONTRACT")
+	if contractAddress == "" {
+		t.Fatal("SUBMISSION_PROXY_CONTRACT not set")
+	}
+
+	tmpHelper, err := helper.NewChainHelper(ctx)
+	if err != nil {
+		t.Fatalf("error creating chain helper: %v", err)
+	}
+	defer tmpHelper.Close()
+
+	whitelist, err := ReadOnchainWhitelist(ctx, tmpHelper, contractAddress, GET_ONCHAIN_WHITELIST)
+	if err != nil {
+		t.Fatalf("error reading onchain whitelist: %v", err)
+	}
+
+	for groupInterval, pairs := range groupedSubmissionPairs {
+		_, err := NewReporter(
+			ctx,
+			WithConfigs(pairs),
+			WithInterval(groupInterval),
+			WithContractAddress(contractAddress),
+			WithCachedWhitelist(whitelist),
+			WithKaiaHelper(tmpHelper),
+			WithLatestData(app.LatestData),
+		)
+		if err != nil {
+			t.Fatalf("error creating new reporter: %v", err)
+		}
+	}
+}
+
+func TestShouldReportDeviation(t *testing.T) {
+	ctx := context.Background()
+	app := New()
+
+	err := app.setReporters(ctx)
+	if err != nil {
+		t.Fatalf("error setting reporters: %v", err)
+	}
+
+	assert.False(t, ShouldReportDeviation(0, 0, 0.05))
+	assert.True(t, ShouldReportDeviation(0, 100000000, 0.05))
+	assert.False(t, ShouldReportDeviation(100000000000, 100100000000, 0.05))
+	assert.True(t, ShouldReportDeviation(100000000000, 105100000000, 0.05))
+	assert.False(t, ShouldReportDeviation(100000000000, 0, 0.05))
+}
+
+func TestGetDeviationThreshold(t *testing.T) {
+	assert.Equal(t, 0.05, GetDeviationThreshold(15*time.Second))
+	assert.Equal(t, 0.01, GetDeviationThreshold(60*time.Minute))
+	assert.Equal(t, 0.05, GetDeviationThreshold(1*time.Second))
+	assert.Equal(t, 0.01, GetDeviationThreshold(2*time.Hour))
+	assert.Less(t, GetDeviationThreshold(30*time.Minute), 0.05)
+}
```

Comment:
let's update the logic, how about 1% threshold for 1 minute? how do you think? since max submission interval is now 1min

Reply:
yep sounds good, will add that option as well

Reply:
Awh I was meant for `GetDeviationThreshold` it is now mapped to 15 sec -> 5% & 1hr -> 1% but I was suggesting an update